db.go

```diff
@@ -0,0 +1,97 @@
+package money
+
+import (
+	"database/sql/driver"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Scan implements sql.Scanner to deserialize a Money instance from an "amount,currency_code" comma-separated pair
+func (m *Money) Scan(src interface{}) error {
+	amount:= &Amount{}
+	currency:= &Currency{}
+
+	// let's support string and int64
+	switch src.(type) {
+	case string:
+		parts := strings.Split(src.(string), ",")
+		if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+			return fmt.Errorf("%#v is not valid to scan into Money; update your query to return a comma-separated pair of \"amount,currency_code\"", src.(string))
+		}
+		if err := amount.Scan(parts[0]); err != nil {
+			return fmt.Errorf("scanning %#v into an Amount: %v", parts[0], err)
+		}
+		if err := currency.Scan(parts[1]); err != nil {
+			return fmt.Errorf("scanning %#v into a Currency: %v", parts[1], err)
+		}
+	default:
+		return fmt.Errorf("don't know how to scan %T into Money; update your query to return a comma-separated pair of \"amount,currency_code\"", src)
+	}
+
+	// allocate new Money with the scanned amount and currency
+	*m = Money{
+		amount: amount,
+		currency: currency,
+	}
+
+	return nil
+}
+
+// Value implements driver.Valuer to serialize an Amount into sql/driver compatible type
+func (a Amount) Value() (driver.Value, error) {
+	return a.val, nil
+}
+
+// Scan implements sql.Scanner to deserialize an Amount from a string or integer value read from a database
+func (a *Amount) Scan(src interface{}) error {
+	var aa Amount
+	// let's support string and int64
+	switch src.(type) {
+	case string:
+		if i, err := strconv.ParseInt(src.(string), 10, 64); err != nil {
```

Comment:
My doubt here is the string representation of an amount should have the decimal separator or not.
Introducing the decimal separator in the string representation of an amount has the benefit of being explicit on it, but has the drawback of "," is a valid decimal separator for some currencies.

Reply:
this is easy to change.

we might use a pipe (`|`) or an at (`@`) or a colon (`:`) or semi-colon (`;`)

or even could implement this value separator as a package level variable (for example `money.DBValueSeparator`) with a default value and allow the client to override it as needed.

Reply:
refactored this out and made it configurable, updated the default to a pipe (`|`)